format/index.md

```diff
@@ -52,636 +58,370 @@ called out below.
-  }
+  "type": "posts",
+  "id": "1",
+  "title": "Rails is Omakase"
 }
+// ...
 ```
 
-In the example above, the post's resource object is simply:
-
-```javascript
-//...
-  {
-    "id": "1",
-    "title": "Rails is Omakase"
-  }
-//...
-```
-
-This section will focus exclusively on resource objects, outside of the context
-of a full JSON API document.
-
 #### Resource Attributes <a href="#document-structure-resource-object-attributes" id="document-structure-resource-object-attributes" class="headerlink"></a>
 
-There are four reserved keys in resource objects:
+A resource object **MUST** contain at least the following top-level members:
 
 * `"id"`
 * `"type"`
-* `"href"`
-* `"links"`
-
-Every other key in a resource object represents an "attribute". An attribute's
-value may be any JSON value.
-
-#### Resource IDs <a href="#document-structure-resource-object-ids" id="document-structure-resource-object-ids" class="headerlink"></a>
-
-Each resource object **SHOULD** contain a unique identifier, or ID, when
-available. IDs **MAY** be assigned by the server or by the client, as described
-below, and **SHOULD** be unique for a resource when scoped by its type. An ID
-**SHOULD** be represented by an `"id"` key and its value **MUST** be a string
-which **SHOULD** only contain alphanumeric characters, dashes and underscores.
-
-IDs can be used with URL templates to fetch related resources, as described
-below.
 
-In scenarios where uniquely identifying information between client and server
-is unnecessary (e.g. read-only, transient entities), JSON API allows for
-omitting IDs.
+In addition, a resource object **MAY** contain any of these top-level members:
 
-#### Resource Types <a href="#document-structure-resource-types" id="document-structure-resource-types" class="headerlink"></a>
-
-The type of each resource object can usually be determined from the context in
-which it is contained. As discussed above, resource objects are typically keyed
-by their type in a document.
-
-Each resource object **MAY** contain a `"type"` key to explicitly designate its
-type.
-
-The `"type"` key is **REQUIRED** when the type of a resource is not otherwise
-specified in a document.
+* `"links"`: information about a resource's relationships (described
+  below).
+* `"meta"`: non-standard meta-information about a resource that can not be
+  represented as an attribute or relationship.
 
-#### Resource URLs <a href="#document-structure-resource-urls" id="document-structure-resource-urls" class="headerlink"></a>
+Any other member in a resource object represents an "attribute", which may
+contain any valid JSON value.
 
-The URL of each resource object **MAY** be specified with the `"href"` key.
-Resource URLs **SHOULD** only be specified by the server and therefore are
-typically only included in response documents.
+Attribute names **MUST** consist of only lower case alphanumeric characters
+and dashes (U+002D: HYPHEN-MINUS, "-"). Attribute names **MUST NOT** begin
```

Comment:
> As for our decision to go with dasherized attributes - we wanted to avoid requiring out-of-band knowledge to parse a response @dgeb 

*Out-of-band knowledge*? How's dashes better? In 9 years of work, I've never seen an API using dashes, not even docs mentioning it. I really don't understand this argument. Could you elaborate?